app/controllers/users_controller.rb

```diff
@@ -120,7 +120,7 @@ def admin_update_params
     end
 
     def generate_random_password
-      token = Devise.friendly_token 32
+      token = Devise.friendly_token User::OMNIAUTH_PASSWORD_LENGTH
```

Comment:
what does friendly mean here? Are there also cryptic tokens?

Reply:
As far as i know, that generates a base64 string, which can be saved as ASCII, but is random.